crates/ruff/src/checkers/ctx.rs

```diff
@@ -0,0 +1,252 @@
+use nohash_hasher::IntMap;
+use rustc_hash::FxHashMap;
+use rustpython_parser::ast::{Expr, Stmt};
+use smallvec::smallvec;
+
+use ruff_python::typing::TYPING_EXTENSIONS;
+
+use crate::ast::helpers::{collect_call_path, from_relative_import};
+use crate::ast::types::{Binding, BindingKind, CallPath, ExecutionContext, RefEquality, Scope};
+use crate::settings::Settings;
+use crate::visibility::VisibleScope;
+
+#[allow(clippy::struct_excessive_bools)]
+pub struct AstContext<'a> {
+    pub settings: &'a Settings,
+    pub module_path: Option<Vec<String>>,
```

Comment:
Nit: `Vec<String>`. An empty `vec![]` is the same as `None` (but requires less space)

Reply:
(I'll change this on `main`, since this is just a move.)